src/components/Payment/Payment.js

```diff
@@ -35,34 +35,49 @@ class Payment extends Component {
   }
 
   async componentDidMount() {
-    const { t } = this.props;
-    try {
-      const response = await getPaymentMethods();
-      const { paymentMethods } = response.responseData;
-      if (paymentMethods) {
-        if (!paymentMethods.length) {
-          this.setState({
-            generalError: t('Payment methods are not defined')
-          });
-        } else {
+    const { t, availablePaymentMethods } = this.props;
+    if (availablePaymentMethods && availablePaymentMethods.length) {
+      this.setState({
+        paymentMethods: availablePaymentMethods
```

Comment:
we should validate the availablePaymentMethods props -> if it has all required data and if the passes paymentMethods are supported. If validation fails, we can throw an error to the console and fetch the payment details from the API. We should still make sure to show only Adyen and PayPal